app/main.py

```diff
@@ -88,6 +90,41 @@ def _get_client(self):
                                              self.client_id, self.client_secret)
         return KeyVaultClient(credentials)
 
+    def _get_tenant_id(self, tenant_id_from_config):
+        if os.getenv('AUTO_DETECT_AAD_TENANT').lower() != 'true':
+            _logger.info('AAD tenant auto detection turned off. Using tenant id %s from cloud config', tenant_id_from_config)
+            return tenant_id_from_config
+        
+        # if we are unable to auto detect tenant id for any reason, we will use the one from config
+        try:
+            vault_base_url = os.getenv('VAULT_BASE_URL')
```

Comment:
Looks like we don't document this in README.md. Can we add an entry for this?

Reply:
addressed.